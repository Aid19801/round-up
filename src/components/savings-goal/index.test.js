import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import SavingsGoal from './index';
import Title from '../../components/title';
import store from '../../store';

configure({ adapter: new Adapter() })

describe('Savings Goal', () => {
  
  it('should render the Savings container without exploding...', () => {
      const wrapper = shallow(<SavingsGoal />);
      expect(wrapper).toBeTruthy();
      expect(wrapper.contains(<Title />))
  });
  
  it('matches the snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
            <SavingsGoal />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  })

  it.skip('should render the expected children', () => {
    const el = <h4 className="loading">loading...</h4>;
    const wrapper = shallow(<SavingsGoal isLoading />);
    expect(wrapper.find(el)).toHaveLength(1);
  });

  it.skip('should render the expected children', () => {
    const el = <h4>an error has occurred...</h4>;
    const wrapper = shallow(<SavingsGoal error />);
    expect(wrapper.find(el)).toHaveLength(1);
  });

});