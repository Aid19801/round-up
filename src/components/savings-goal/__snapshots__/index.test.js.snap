// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Savings Goal matches the snapshot 1`] = `
<div
  className="col-sm-6 savings-container"
>
  <h4>
    Savings Goal: £
    0
  </h4>
  <h4>
    current savings: £
    0
  </h4>
  <input
    className="goal-input"
    name="savingsGoal"
    onChange={[Function]}
    placeholder="enter your savings goal here..."
    type="text"
  />
  <button
    onClick={[Function]}
  >
    submit
  </button>
</div>
`;
